tests/bitcoin/BitcoinClient.spec.ts

```diff
@@ -1100,6 +1102,44 @@ describe('BitcoinClient', async () => {
       done();
     });
 
+    it('should call retry-fetch without authohrization', async (done) => {
```

Comment:
Just a comment: I am also inclined to say the case of not having authorization is unnecessary bloat as from memory Bitcoin Client requires auth for you to use its API, and this is trying to handle a a hypothetical case where an alternative implementation does not require it.